ecommerce/backend/API/views/vendor.py

```diff
@@ -14,3 +14,23 @@ def vendor_details(request,vendor_id):
     if vendor is None:
         return Response(status=status.HTTP_400_BAD_REQUEST)
     return Response({'title':vendor.title,'image_url':vendor.image_url,'first_name':vendor.first_name,'last_name':vendor.last_name,'last_name':vendor.last_name,'total_rating':vendor.total_rating,'rating_count':vendor.rating_count,'description':vendor.description})
+
+@api_view(['PUT'])
+@permission_classes([permissions.IsVendorUser]) 
+def vendor_profile(request):
+    vendor = User.objects.filter(id=request.user.id).first()
+    if "image" in request.data:
+        Vendor.objects.filter(user_id=vendor.id).update(title=request.data["title"],description=request.data["description"])
+    else:
+        img_array = base64.b64decode(request.data["image"])
+        image = Image(image=img_array)
+        image.save()
+        image_url="/image/"+str(image.pk)
+        Vendor.objects.filter(user_id=vendor.id).update(title=request.data["title"],description=request.data["description"],image_url=image_url)
```

Comment:
You control if image in request.data, but in else statement, you take request.data["image"].  There is a conflict.

Reply:
Fixed